### ISSUE TITLE
Private superclasses lead to dead references

### ISSUE THREAD

**Issue by Hixie** [dart-lang/dartdoc#1476]
Consider this setup:

```dart
class _Private {
  /// This is the documentation for "foo".
  final bool foo;
}

/// See the [foo] documentation.
class Public extends _Private { }
```

You'll get the error message (paraphrasing a little) "warning: no canonical library found for test._Private.foo, not linking from _Private.foo referred to by Public".

But `Public.foo` appears in the documentation with the text from `_Private.foo`, so it seems like `Public` should just link to its own `foo` when it refers to `[foo]`.

Marking P1 as "Broken links, few or on edge cases".

**Comment by jcollins-g** [comment]
This is related to #1173, but probably fixable independent of that.

**Comment by jcollins-g** [comment]
Fixed in #1524.
